### PR TITLE
Closes #2501-general-issue-make-subject-keys-flexible-in-all-functions

### LINKED ISSUE #2501: General Issue: Make subject keys flexible in all functions
### Background Information

@bundfussr did a review of the admiral functions and found 2 occurrences where `STUDYID` and `USUBJID` are hardcoded instead of using `get_admiral_option("subject_keys")` as per https://pharmaverse.github.io/admiral/reference/#admiral-options.

These are `derive_vars_atc()` and `create_single_dose_dataset()`, for example:

```
derive_vars_atc <- function(dataset,
                            dataset_facm,
                            by_vars = exprs(USUBJID, CMREFID = FAREFID),
                            value_var = FASTRESC)
```

### Definition of Done

These functions should be flexible so as if ever `set_admiral_options(subject_keys = exprs(STUDYID, USUBJID2))` was set by user then these variables would be used everywhere instead of `USUBJID`.

### PR DESCRIPTION
**Pull request by ProfessorP-beep**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by ProfessorP-beep — COMMENTED** [review]
I had an issue when doing pkgdown::build_site() with the bds_exposure script giving an error for line 365. Tried to update that too but its still throwing back an error. I'm pushing the changes for now while I sort out review errors.

**Comment by ProfessorP-beep** [comment]
> I had an issue when doing pkgdown::build_site() with the bds_exposure script giving an error for line 365. Tried to update that too but its still throwing back an error. I'm pushing the changes for now while I sort out review errors.

I figured it out. I just had to update Rtools, and finding where errors in the test scripts used the updated functions was easier.

File: NEWS.md [hunk 1]
```diff
@@ -45,7 +45,8 @@
 </details>
 
 # admiral 1.1.1
-
+- `derive_vars_atc()` and `create_single_dose_dataset()` `by_vars` argument updated to use `get_admiral_option("subject_keys")` instead of  `USUBJID` or `STUDYID`.
```
Comment (bms63) [inline NEWS.md:48]: These updates should go under the development version section of the news.  When we release it will change to admiral 1.1.2

Comment (ProfessorP-beep) [reply NEWS.md:48]: Done! Pushing now.

File: vignettes/bds_exposure.Rmd [hunk 2]
```diff
@@ -361,7 +361,7 @@ For example, to calculate the total dose by subject and treatment,
 adex <- derive_param_exposure(
   adex,
   dataset_add = adex,
-  by_vars = exprs(STUDYID, USUBJID, !!!adsl_vars),
+  by_vars = c(get_admiral_option("subject_keys"), adsl_vars),
```
Comment (bundfussr) [inline vignettes/bds_exposure.Rmd:364]: Either we should update all calls in the vignette or none.

@bms63 , did we reach a decision regarding updating vignettes and templates?

**Comment by ProfessorP-beep** [comment]
Sorry, I'll get this fixed by the weekend. Work got hectic and preparing for an interview!

File: NEWS.md [hunk 3]
```diff
@@ -10,6 +10,10 @@
 - Replace use of `data("sdtm")` with `sdtm <- pharmaverse::sdtm` in templates and vignettes. (#2498)
 
 ## Updates of Existing Functions
+- `derive_vars_atc()` and `create_single_dose_dataset()` `by_vars` argument updated to use `get_admiral_option("subject_keys")` instead of  `USUBJID` or `STUDYID`.
+- test scripts, R, and markdown files for `create_single_dose_dataset` and `occds.Rmd` updated to include a `STUDYID` column because of `get_admiral_option("subject_keys")` update above. 
+- `derive_extreme_event()` was fixed such that `check_type = "none"` is accepted
+again. (#2462)
```
[the remark this answers is not in the document]
Comment (ProfessorP-beep) [reply NEWS.md:16]: Sorry, 1.1.0 or 1.1.1?

File: tests/testthat/test-create_single_dose_dataset.R [hunk 4]
```diff
@@ -2,24 +2,24 @@
...
-    ~USUBJID, ~EXDOSFRQ,       ~ASTDT,            ~AENDT,
-    "P01",    "Q2D",           ymd("2021-01-01"), ymd("2021-01-07"),
-    "P01",    "Q3D",           ymd("2021-01-08"), ymd("2021-01-14"),
-    "P01",    "EVERY 2 WEEKS", ymd("2021-01-15"), ymd("2021-01-29"),
-    "P02",    "ONCE",          ymd("2021-02-02"), ymd("2021-02-02")
+    ~STUDYID, ~USUBJID, ~EXDOSFRQ, ~ASTDT, ~AENDT,
+    "STUDY01", "P01", "Q2D", ymd("2021-01-01"), ymd("2021-01-07"),
+    "STUDY01", "P01", "Q3D", ymd("2021-01-08"), ymd("2021-01-14"),
+    "STUDY01", "P01", "EVERY 2 WEEKS", ymd("2021-01-15"), ymd("2021-01-29"),
+    "STUDY01", "P02", "ONCE", ymd("2021-02-02"), ymd("2021-02-02")
```
[the remark this answers is not in the document]
Comment (ProfessorP-beep) [reply tests/testthat/test-create_single_dose_dataset.R:9]: Fixing all these

File: vignettes/bds_exposure.Rmd (hunk 2 above)
Comment (ProfessorP-beep) [reply vignettes/bds_exposure.Rmd:364]: I went ahead an replaced all the incidents in this vignette until we decide to replace all the others.

**Comment by ProfessorP-beep** [comment]
I just noticed that after I run styler is misaligns the tables in the test scripts again. I can correct and we can just accept the failed styler check?

File: NEWS.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (bundfussr) [reply NEWS.md]: This was implemented in 1.1.1. See https://pharmaverse.github.io/admiral/news/index.html.

**Comment by bundfussr** [comment]
> I just noticed that after I run styler it misaligns the tables in the test scripts again. I can correct and we can just accept the failed styler check?

Usually styler preserves alignment in tables if
- all columns including the header are left or right aligned and
- the first column is left aligned.

I think when you aligned the tables you centered most of the columns. This is not considered as aligned by styler.

**Comment by bms63** [comment]
I fixed the news here - noticed something else got messed up from another PR with the news.

@bundfussr was there anything else that we needed to fix here?

**Comment by bundfussr** [comment]
> I fixed the news here - noticed something else got messed up from another PR with the news.
> 
> @bundfussr was there anything else that we needed to fix here?

@bms63 , the news are fine now.

Just the alignment of the `tribble()` calls is pending.

**Comment by bms63** [comment]
@bundfussr I think I am confused by styler request.  I have run styler and no changes are implemented for alignment

image

I think styler wants to put in a space after the comma so we can't align the data to the header anyways.  Let me know what I am missing, but if you have the time can you just fix this how you want it as I guess I am not following.

**Comment by bundfussr** [comment]
> @bundfussr I think I am confused by styler request. I have run styler and no changes are implemented for alignment
> 
> image
> 
> I think styler wants to put in a space after the comma so we can't align the data to the header anyways as the studyid takes up more room now. Let me know what I am missing, but if you have the time can you just fix this how you want it as I guess I am not following.

@bms63 , I've aligned the `tribble()` calls.

**Comment by bundfussr** [comment]
> So you want character values manually left aligned and numeric values right aligned?

That would be ideal for readability.
> 
> Wondering if we automate this as I am never going to remember to do this?

This would be really helpful but would require some work. An addin which aligns a selected `tribble()` call would be nice.

**Comment by bms63** [comment]
> > So you want character values manually left aligned and numeric values right aligned?
> 
> That would be ideal for readability.
> 
> > Wondering if we automate this as I am never going to remember to do this?
> 
> This would be really helpful but would require some work. An addin which aligns a selected `tribble()` call would be nice.

@sadchla-codes sounds like a great candidate for https://github.com/pharmaverse/pharmaverse4devs
